mlir/tools/miopen-gen/miopen-gen.cpp

```diff
@@ -690,141 +690,31 @@ static FuncOp createGPUWrapper(ModuleOp &module, const KernelIF &kernel) {
         loc, b.create<arith::ConstantIntOp>(loc, deviceNum.getValue(),
                                             b.getIntegerType(32)));
 
-  // Emit some constant values for HIP runtime API calls.
-  auto oneConstantI32Op =
-      b.create<arith::ConstantIntOp>(loc, 1, b.getIntegerType(32));
-  auto twoConstantI32Op =
-      b.create<arith::ConstantIntOp>(loc, 2, b.getIntegerType(32));
-
-  auto getGpuAllocFunc = [&](mlir::Type elemType) -> FuncOp {
-    StringRef allocFuncName = "mgpuMemAlloc5DFloat";
-    if (elemType.isF32()) {
-    } else if (elemType.isF16()) {
-      allocFuncName = "mgpuMemAlloc5DHalf";
-    } else if (elemType.isBF16()) {
-      allocFuncName = "mgpuMemAlloc5DBF16";
-    } else if (elemType.isInteger(8)) {
-      allocFuncName = "mgpuMemAlloc5DInt8";
-    } else if (elemType.isInteger(32)) {
-      allocFuncName = "mgpuMemAlloc5DInt32";
-    }
-    auto unk5DType = MemRefType::get({-1, -1, -1, -1, -1}, elemType);
-    return makeFuncDecl(module, allocFuncName, {unk5DType}, {unk5DType});
-  };
-
-  auto getGpuDeallocFunc = [&](mlir::Type elemType) -> FuncOp {
-    StringRef deallocFuncName = "mgpuMemDealloc5DFloat";
-    if (elemType.isF32()) {
-    } else if (elemType.isF16()) {
-      deallocFuncName = "mgpuMemDealloc5DHalf";
-    } else if (elemType.isBF16()) {
-      deallocFuncName = "mgpuMemDealloc5DBF16";
-    } else if (elemType.isInteger(8)) {
-      deallocFuncName = "mgpuMemDealloc5DInt8";
-    } else if (elemType.isInteger(32)) {
-      deallocFuncName = "mgpuMemDealloc5DInt32";
-    }
-    auto unk5DType = MemRefType::get({-1, -1, -1, -1, -1}, elemType);
-    return makeFuncDecl(module, deallocFuncName, {unk5DType});
-  };
-
-  auto getGpuCopyFunc = [&](mlir::Type elemType) -> FuncOp {
-    StringRef allocFuncName = "mgpuMemCopy5DFloat";
-    if (elemType.isF32()) {
-    } else if (elemType.isF16()) {
-      allocFuncName = "mgpuMemCopy5DHalf";
-    } else if (elemType.isBF16()) {
-      allocFuncName = "mgpuMemCopy5DBF16";
-    } else if (elemType.isInteger(8)) {
-      allocFuncName = "mgpuMemCopy5DInt8";
-    } else if (elemType.isInteger(32)) {
-      allocFuncName = "mgpuMemCopy5DInt32";
```

Comment:
If no other use cases, get rid of those `mgpu*` functions?

Reply:
I'd go make sure of what exactly the `gpu.*` ops lower to

Reply:
The gpu.* funcs will lower to mgpu* funcs, so they are still needed.

Reply:
Sorry, the 5D funcs are not needed..